src/js/tools/superselect.js

```diff
@@ -74,25 +75,27 @@ define(function (require, exports, module) {
                         UI.pointerPropagationMode.ALPHA_PROPAGATE_WITH_NOTIFY);
                 });
         } else {
-            var currentLayers = currentDocument.layers.selected,
-                currentLayer = currentLayers.first();
-            
-            if (currentLayer.vectorMaskEnabled && !currentLayer.vectorMaskEmpty) {
-                return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing())
-                    .bind(this)
-                    .then(function () {
-                        var pointerPolicy = new PointerEventPolicy(UI.policyAction.ALWAYS_PROPAGATE,
-                            OS.eventKind.LEFT_MOUSE_DOWN);
-                        return this.transfer(policy.addPointerPolicies, [pointerPolicy]);
-                    })
-                    .then(function () {
-                        // We are not transfering here, because we activly want to end the use of our locks
-                        this.flux.actions.tools.enterPathModalState();
-                    });
-            } else {
-                var pointerPolicy = new PointerEventPolicy(UI.policyAction.ALWAYS_PROPAGATE,
-                    OS.eventKind.LEFT_MOUSE_DOWN);
-                return this.transfer(policy.addPointerPolicies, [pointerPolicy]);
+            var currentLayers = currentDocument.layers.selected;
+
+            if (currentLayers) {
```

Comment:
`currentLayers` will always be truthy. You need to explicitly test it's size. How about:
```
if (currentLayers.isEmpty()) {
    return;
}
```